src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -54,6 +54,36 @@ module Reader = struct
     assert_not_read reader ;
     reader.has_reader <- true ;
     wrap_reader (Pipe.filter_map reader.reader ~f)
+
+  let iter_sync ?consumer ?continue_on_error reader ~f =
```

Comment:
What is `_sync` indicating here? Just that `f` doesn't return a `'a Deferred.t`? Does this actually make the underlying `Pipe.fold` operate synchronously? I don't think it does, but could be wrong...

Reply:
Yes, `_sync` just refers to the dispatch being synchronous. I can rename if `_sync` is too unclear.

Reply:
Both of these `iter` functions are actually sequential. Even the one that returns deferreds for each read blocks on that deferred to provide pushback to the writer before reading again. There are other specialized versions you use to handle multiple in parallel (not implemented in `Strict_pipe` right now).